# <span style="color:red">EXERCICE 1</span>

# <span style="color:blue">Étape 1</span>

# 1)Charger les données

In [35]:
import numpy as nf
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [36]:
df_series= pd.read_csv('EdStatsCountry-Series.csv')
df_country=pd.read_csv('EdStatsCountry.csv')
df_data= pd.read_csv('EdStatsData.csv')
df_foot= pd.read_csv('EdStatsFootNote.csv')
df_stat_series= pd.read_csv('EdStatsSeries.csv')

# 2)Visualiser dans un Jupyter Notebook les premières lignes des cinq fichiers en utilisant Pandas

In [12]:
df_series.head()

CountryCode         SeriesCode  \
0         ABW        SP.POP.TOTL   
1         ABW        SP.POP.GROW   
2         AFG        SP.POP.GROW   
3         AFG  NY.GDP.PCAP.PP.CD   
4         AFG        SP.POP.TOTL   

                                         DESCRIPTION  Unnamed: 3  
0  Data sources : United Nations World Population...         NaN  
1  Data sources: United Nations World Population ...         NaN  
2  Data sources: United Nations World Population ...         NaN  
3                 Estimates are based on regression.         NaN  
4  Data sources : United Nations World Population...         NaN

In [11]:
df_country.head()

Country Code   Short Name   Table Name                     Long Name  \
0          ABW        Aruba        Aruba                         Aruba   
1          AFG  Afghanistan  Afghanistan  Islamic State of Afghanistan   
2          AGO       Angola       Angola   People's Republic of Angola   
3          ALB      Albania      Albania           Republic of Albania   
4          AND      Andorra      Andorra       Principality of Andorra   

  2-alpha code   Currency Unit  \
0           AW   Aruban florin   
1           AF  Afghan afghani   
2           AO  Angolan kwanza   
3           AL    Albanian lek   
4           AD            Euro   

                                       Special Notes  \
0  SNA data for 2000-2011 are updated from offici...   
1  Fiscal year end: March 20; reporting period fo...   
2  April 2013 database update: Based on IMF data,...   
3                                                NaN   
4                                                NaN   

                      Region          Income Group WB-2 code  ...  \
0  Latin America & Caribbean  High income: nonOECD        AW  ...   
1                 South Asia            Low income        AF  ...   
2         Sub-Saharan Africa   Upper middle income        AO  ...   
3      Europe & Central Asia   Upper middle income        AL  ...   
4      Europe & Central Asia  High income: nonOECD        AD  ...   

            IMF data dissemination standard  \
0                                       NaN   
1  General Data Dissemination System (GDDS)   
2  General Data Dissemination System (GDDS)   
3  General Data Dissemination System (GDDS)   
4                                       NaN   

                            Latest population census  \
0                                               2010   
1                                               1979   
2                                               1970   
3                                               2011   
4  2011. Population figures compiled from adminis...   

                             Latest household survey  \
0                                                NaN   
1  Multiple Indicator Cluster Survey (MICS), 2010/11   
2               Malaria Indicator Survey (MIS), 2011   
3       Demographic and Health Survey (DHS), 2008/09   
4                                                NaN   

   Source of most recent Income and expenditure data  \
0                                                NaN   
1            Integrated household survey (IHS), 2008   
2            Integrated household survey (IHS), 2008   
3  Living Standards Measurement Study Survey (LSM...   
4                                                NaN   

  Vital registration complete Latest agricultural census  \
0                         Yes                        NaN   
1                         NaN                    2013/14   
2                         NaN                       2015   
3                         Yes                       2012   
4                         Yes                        NaN   

  Latest industrial data Latest trade data Latest water withdrawal data  \
0                    NaN            2012.0                          NaN   
1                    NaN            2012.0                         2000   
2                    NaN               NaN                         2005   
3                 2010.0            2012.0                         2006   
4                    NaN            2006.0                          NaN   

  Unnamed: 31  
0         NaN  
1         NaN  
2         NaN  
3         NaN  
4         NaN  

[5 rows x 32 columns]

In [13]:
df_data.head()

Country Name Country Code  \
0   Arab World          ARB   
1   Arab World          ARB   
2   Arab World          ARB   
3   Arab World          ARB   
4   Arab World          ARB   

                                      Indicator Name  Indicator Code  \
0  Adjusted net enrolment rate, lower secondary, ...      UIS.NERA.2   
1  Adjusted net enrolment rate, lower secondary, ...    UIS.NERA.2.F   
2  Adjusted net enrolment rate, lower secondary, ...  UIS.NERA.2.GPI   
3  Adjusted net enrolment rate, lower secondary, ...    UIS.NERA.2.M   
4  Adjusted net enrolment rate, primary, both sex...     SE.PRM.TENR   

        1970       1971       1972       1973       1974      1975  ...  2060  \
0        NaN        NaN        NaN        NaN        NaN       NaN  ...   NaN   
1        NaN        NaN        NaN        NaN        NaN       NaN  ...   NaN   
2        NaN        NaN        NaN        NaN        NaN       NaN  ...   NaN   
3        NaN        NaN        NaN        NaN        NaN       NaN  ...   NaN   
4  54.822121  54.894138  56.209438  57.267109  57.991138  59.36554  ...   NaN   

   2065  2070  2075  2080  2085  2090  2095  2100  Unnamed: 69  
0   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
1   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
2   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
3   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
4   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  

[5 rows x 70 columns]

In [14]:
df_foot.head()

CountryCode      SeriesCode    Year          DESCRIPTION  Unnamed: 4
0         ABW  SE.PRE.ENRL.FE  YR2001  Country estimation.         NaN
1         ABW  SE.TER.TCHR.FE  YR2005  Country estimation.         NaN
2         ABW  SE.PRE.TCHR.FE  YR2000  Country estimation.         NaN
3         ABW  SE.SEC.ENRL.GC  YR2004  Country estimation.         NaN
4         ABW     SE.PRE.TCHR  YR2006  Country estimation.         NaN

In [15]:
df_stat_series.head()

Series Code       Topic  \
0  BAR.NOED.1519.FE.ZS  Attainment   
1     BAR.NOED.1519.ZS  Attainment   
2  BAR.NOED.15UP.FE.ZS  Attainment   
3     BAR.NOED.15UP.ZS  Attainment   
4  BAR.NOED.2024.FE.ZS  Attainment   

                                      Indicator Name  \
0  Barro-Lee: Percentage of female population age...   
1  Barro-Lee: Percentage of population age 15-19 ...   
2  Barro-Lee: Percentage of female population age...   
3  Barro-Lee: Percentage of population age 15+ wi...   
4  Barro-Lee: Percentage of female population age...   

                                    Short definition  \
0  Percentage of female population age 15-19 with...   
1  Percentage of population age 15-19 with no edu...   
2  Percentage of female population age 15+ with n...   
3  Percentage of population age 15+ with no educa...   
4  Percentage of female population age 20-24 with...   

                                     Long definition  Unit of measure  \
0  Percentage of female population age 15-19 with...              NaN   
1  Percentage of population age 15-19 with no edu...              NaN   
2  Percentage of female population age 15+ with n...              NaN   
3  Percentage of population age 15+ with no educa...              NaN   
4  Percentage of female population age 20-24 with...              NaN   

  Periodicity Base Period Other notes Aggregation method  ...  \
0         NaN         NaN         NaN                NaN  ...   
1         NaN         NaN         NaN                NaN  ...   
2         NaN         NaN         NaN                NaN  ...   
3         NaN         NaN         NaN                NaN  ...   
4         NaN         NaN         NaN                NaN  ...   

  Notes from original source  General comments  \
0                        NaN               NaN   
1                        NaN               NaN   
2                        NaN               NaN   
3                        NaN               NaN   
4                        NaN               NaN   

                                              Source  \
0  Robert J. Barro and Jong-Wha Lee: http://www.b...   
1  Robert J. Barro and Jong-Wha Lee: http://www.b...   
2  Robert J. Barro and Jong-Wha Lee: http://www.b...   
3  Robert J. Barro and Jong-Wha Lee: http://www.b...   
4  Robert J. Barro and Jong-Wha Lee: http://www.b...   

  Statistical concept and methodology Development relevance  \
0                                 NaN                   NaN   
1                                 NaN                   NaN   
2                                 NaN                   NaN   
3                                 NaN                   NaN   
4                                 NaN                   NaN   

  Related source links Other web links  Related indicators  License Type  \
0                  NaN             NaN                 NaN           NaN   
1                  NaN             NaN                 NaN           NaN   
2                  NaN             NaN                 NaN           NaN   
3                  NaN             NaN                 NaN           NaN   
4                  NaN             NaN                 NaN           NaN   

   Unnamed: 20  
0          NaN  
1          NaN  
2          NaN  
3          NaN  
4          NaN  

[5 rows x 21 columns]

# <span style="color:blue">Étape 2</span>

## 1)Définir ce que représente une ligne: Calculez le nombre de lignes et de colonnes
## 2) Calculez le nombre de doublons dans le jeu de données.
## 3) Supprimez les doublons s’il y en a.
## 4) Calculez la proportion de valeurs manquantes par colonne.
## 5) Supprimez les colonnes inutilisables.
## 6) Pour les colonnes numériques : calculez les statistiques descriptives basiques en utilisant describe().
## 7) Pour les colonnes catégorielles : calculez le nombre d'occurrences de chaque valeur possible de la colonne.

### <span style="color:violet"> DATA FRAME SERIES</span>

In [4]:
df_series.columns

Index(['CountryCode', 'SeriesCode', 'DESCRIPTION', 'Unnamed: 3'], dtype='object')

In [5]:
df_data.columns

Index(['Country Name', 'Country Code', 'Indicator Name', 'Indicator Code',
       '1970', '1971', '1972', '1973', '1974', '1975', '1976', '1977', '1978',
       '1979', '1980', '1981', '1982', '1983', '1984', '1985', '1986', '1987',
       '1988', '1989', '1990', '1991', '1992', '1993', '1994', '1995', '1996',
       '1997', '1998', '1999', '2000', '2001', '2002', '2003', '2004', '2005',
       '2006', '2007', '2008', '2009', '2010', '2011', '2012', '2013', '2014',
       '2015', '2016', '2017', '2020', '2025', '2030', '2035', '2040', '2045',
       '2050', '2055', '2060', '2065', '2070', '2075', '2080', '2085', '2090',
       '2095', '2100', 'Unnamed: 69'],
      dtype='object')

In [6]:
df_foot.columns

Index(['CountryCode', 'SeriesCode', 'Year', 'DESCRIPTION', 'Unnamed: 4'], dtype='object')

In [7]:
df_stat_series.columns

Index(['Series Code', 'Topic', 'Indicator Name', 'Short definition',
       'Long definition', 'Unit of measure', 'Periodicity', 'Base Period',
       'Other notes', 'Aggregation method', 'Limitations and exceptions',
       'Notes from original source', 'General comments', 'Source',
       'Statistical concept and methodology', 'Development relevance',
       'Related source links', 'Other web links', 'Related indicators',
       'License Type', 'Unnamed: 20'],
      dtype='object')

In [8]:
print(df_series.shape)

(613, 4)


In [9]:
print(df_series.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 613 entries, 0 to 612
Data columns (total 4 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   CountryCode  613 non-null    object 
 1   SeriesCode   613 non-null    object 
 2   DESCRIPTION  613 non-null    object 
 3   Unnamed: 3   0 non-null      float64
dtypes: float64(1), object(3)
memory usage: 19.3+ KB
None


### <span style="color:pink"> Le DF Series contient 613 lignes qui sont numéroté de 0 à 612 et 4 colonnes </span>

### <span style="color:pink"> Object(3): Signifie dire que les 3 premières colonnes contiennent 613 valeurs qui sont non null, soit pas vides avec du texte ou chaine de caractères </span>
### <span style="color:pink"> Dtypes: float64(1): Signifie que la colonne 'unnamed' peux contenir des valeurs décimales mais qu'elle est vide: je choisi de la supprimer tout de suite, ainsi que les descriptions qui me sont d'aucune utilité pour le projet </span>

In [37]:
df_series = df_series.drop(columns=['Unnamed: 3'])
df_series = df_series.drop(columns=['DESCRIPTION'])

In [38]:
df_series.head()

CountryCode         SeriesCode
0         ABW        SP.POP.TOTL
1         ABW        SP.POP.GROW
2         AFG        SP.POP.GROW
3         AFG  NY.GDP.PCAP.PP.CD
4         AFG        SP.POP.TOTL

In [40]:
print(df_series.describe(include='object'))

       CountryCode   SeriesCode
count          613          613
unique         211           21
top            MDA  SP.POP.TOTL
freq            18          211


### <span style="color:pink"> Pour COUNT: Signifie qu'il y a 613 lignes dans chacune des deux colonnes CountryCode et SeriesCode </span>
### <span style="color:pink"> Pour UNIQUE : CountryCode: il y a 211 valeurs différentes, donc 211 pays ou zones distincts, SeriesCode: il y a 21 indicateurs différents </span>
### <span style="color:pink"> Pour TOP: CountryCode: La valeur la plus fréquente est : MDA (Pays Moldavie) et SeriesCode: SP.POP.TOTL, représente la statistique population totale </span>
### <span style="color:pink"> FREQ: signifie que → Le code pays MDA apparaît 18 fois, et SP.POP.TOTL apparaît 211 fois, j'imagine 1 ligne par pays car 211 Country Code </span>
### <span style="color:pink"> Je conclue qu'il me manque les indicateurs statistiques chiffrés dans ce DF, tous ceux listés dans SeriesCode (PIB, POP TOTAL, GROW CROISSANCE etc.)

In [43]:
doublons = df_series.duplicated().sum()
print("Nombre de doublons :", doublons)

Nombre de doublons : 0


In [44]:
doublons_colonnes = df_series.duplicated(subset=['CountryCode', 'SeriesCode']).sum()
print("Nombre de doublons colonne Country et Series :", doublons)

Nombre de doublons colonne Country et Series : 0


### <span style="color:pink"> J'ai fait 2 codes, 1 pour trouver des lignes identiques sur toutes les colonnes et 1 autre avec SUBSET, pour savoir si la combinaison CountryCode + SeriesCode est unique, résultat il n'y a pas de doublons je ne supprime plus rien pour ce DF </span>

In [45]:
total_valeurs_manquantes = df_series.isnull().sum().sum()
print("Total de valeurs manquantes :", total_valeurs_manquantes)

Total de valeurs manquantes : 0


### <span style="color:pink"> Il n'y a pas de valeurs manquantes sur ce DF </span>

In [48]:
occurrences_df_series_SeriesCode = df_series['SeriesCode'].value_counts()
print(occurrences_df_series_SeriesCode)

SeriesCode
SP.POP.TOTL          211
SP.POP.GROW          211
NY.GDP.PCAP.PP.CD     19
NY.GDP.PCAP.PP.KD     19
NY.GNP.PCAP.PP.CD     19
NY.GDP.MKTP.PP.KD     14
NY.GNP.MKTP.PP.CD     14
NY.GDP.MKTP.PP.CD     14
SP.POP.0014.TO.ZS     13
SP.POP.TOTL.FE.ZS     13
SP.POP.TOTL.MA.ZS     13
SP.POP.1564.TO.ZS     13
NY.GNP.PCAP.CD         6
NY.GDP.PCAP.CD         5
NY.GDP.PCAP.KD         5
SP.POP.1564.MA.IN      4
SP.POP.0014.TO         4
SP.POP.1564.TO         4
SP.POP.1564.FE.IN      4
SP.POP.0014.MA.IN      4
SP.POP.0014.FE.IN      4
Name: count, dtype: int64


In [47]:
occurrences_df_series_CountryCode = df_series['CountryCode'].value_counts()
print(occurrences_df_series_CountryCode)

CountryCode
MDA    18
GEO    18
CYP    12
SRB    12
MUS    12
       ..
GRL     2
GTM     2
GUM     2
HKG     2
ZWE     2
Name: count, Length: 211, dtype: int64


### <span style="color:pink"> J'ai compté le nombre d'occurence d'indicateurs pour la colonne Series et les nombres d'occurence de pays de la colonne Country </span>

### <span style="color:violet"> DATA FRAME COUNTRY</span> 

In [10]:
df_country.columns

Index(['Country Code', 'Short Name', 'Table Name', 'Long Name', '2-alpha code',
       'Currency Unit', 'Special Notes', 'Region', 'Income Group', 'WB-2 code',
       'National accounts base year', 'National accounts reference year',
       'SNA price valuation', 'Lending category', 'Other groups',
       'System of National Accounts', 'Alternative conversion factor',
       'PPP survey year', 'Balance of Payments Manual in use',
       'External debt Reporting status', 'System of trade',
       'Government Accounting concept', 'IMF data dissemination standard',
       'Latest population census', 'Latest household survey',
       'Source of most recent Income and expenditure data',
       'Vital registration complete', 'Latest agricultural census',
       'Latest industrial data', 'Latest trade data',
       'Latest water withdrawal data', 'Unnamed: 31'],
      dtype='object')

In [11]:
print(df_country.shape)

(241, 32)


In [12]:
print(df_country.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 241 entries, 0 to 240
Data columns (total 32 columns):
 #   Column                                             Non-Null Count  Dtype  
---  ------                                             --------------  -----  
 0   Country Code                                       241 non-null    object 
 1   Short Name                                         241 non-null    object 
 2   Table Name                                         241 non-null    object 
 3   Long Name                                          241 non-null    object 
 4   2-alpha code                                       238 non-null    object 
 5   Currency Unit                                      215 non-null    object 
 6   Special Notes                                      145 non-null    object 
 7   Region                                             214 non-null    object 
 8   Income Group                                       214 non-null    object 
 9   WB-2 code 

### <span style="color:pink"> Le DF Country contient 241 lignes qui sont numéroté de 0 à 240 et 32 colonnes </span>
### <span style="color:pink"> Object(28): Signifie que 28 colonnes contiennent du texte ou chaine de caractères </span>
### <span style="color:pink">float64(4)Signifie que 4 colonnes contiennent des chiffres décimales </span>
### <span style="color:pink"> On remarque que la colonne'unnamed'est vide: je choisi de la supprimer tout de suite</span>

In [13]:
df_country = df_country.drop(columns=['Unnamed: 31'])

In [14]:
df_country.head()

Country Code   Short Name   Table Name                     Long Name  \
0          ABW        Aruba        Aruba                         Aruba   
1          AFG  Afghanistan  Afghanistan  Islamic State of Afghanistan   
2          AGO       Angola       Angola   People's Republic of Angola   
3          ALB      Albania      Albania           Republic of Albania   
4          AND      Andorra      Andorra       Principality of Andorra   

  2-alpha code   Currency Unit  \
0           AW   Aruban florin   
1           AF  Afghan afghani   
2           AO  Angolan kwanza   
3           AL    Albanian lek   
4           AD            Euro   

                                       Special Notes  \
0  SNA data for 2000-2011 are updated from offici...   
1  Fiscal year end: March 20; reporting period fo...   
2  April 2013 database update: Based on IMF data,...   
3                                                NaN   
4                                                NaN   

                      Region          Income Group WB-2 code  ...  \
0  Latin America & Caribbean  High income: nonOECD        AW  ...   
1                 South Asia            Low income        AF  ...   
2         Sub-Saharan Africa   Upper middle income        AO  ...   
3      Europe & Central Asia   Upper middle income        AL  ...   
4      Europe & Central Asia  High income: nonOECD        AD  ...   

     Government Accounting concept           IMF data dissemination standard  \
0                              NaN                                       NaN   
1  Consolidated central government  General Data Dissemination System (GDDS)   
2     Budgetary central government  General Data Dissemination System (GDDS)   
3     Budgetary central government  General Data Dissemination System (GDDS)   
4                              NaN                                       NaN   

                            Latest population census  \
0                                               2010   
1                                               1979   
2                                               1970   
3                                               2011   
4  2011. Population figures compiled from adminis...   

                             Latest household survey  \
0                                                NaN   
1  Multiple Indicator Cluster Survey (MICS), 2010/11   
2               Malaria Indicator Survey (MIS), 2011   
3       Demographic and Health Survey (DHS), 2008/09   
4                                                NaN   

   Source of most recent Income and expenditure data  \
0                                                NaN   
1            Integrated household survey (IHS), 2008   
2            Integrated household survey (IHS), 2008   
3  Living Standards Measurement Study Survey (LSM...   
4                                                NaN   

  Vital registration complete Latest agricultural census  \
0                         Yes                        NaN   
1                         NaN                    2013/14   
2                         NaN                       2015   
3                         Yes                       2012   
4                         Yes                        NaN   

  Latest industrial data Latest trade data Latest water withdrawal data  
0                    NaN            2012.0                          NaN  
1                    NaN            2012.0                         2000  
2                    NaN               NaN                         2005  
3                 2010.0            2012.0                         2006  
4                    NaN            2006.0                          NaN  

[5 rows x 31 columns]

In [15]:
df_country = df_country.drop(columns=['Special Notes'])

### <span style="color:pink"> Je supprime speciale note qui ne m'est pas utile</span>

In [16]:
print(df_country.describe(include='object'))

       Country Code Short Name Table Name Long Name 2-alpha code  \
count           241        241        241       241          238   
unique          241        241        241       241          238   
top             ABW      Aruba      Aruba     Aruba           AW   
freq              1          1          1         1            1   

       Currency Unit                 Region         Income Group WB-2 code  \
count            215                    214                  214       240   
unique           152                      7                    5       240   
top             Euro  Europe & Central Asia  Upper middle income        AW   
freq              23                     57                   55         1   

       National accounts base year  ... External debt Reporting status  \
count                          205  ...                            124   
unique                          43  ...                              3   
top                           2005  ...       

### <span style="color:pink"> J'utilise ce code pour décrire les donnnés 'object'</span>
### <span style="color:pink"> Il y a 241 pays différents comme dans la DF Series</span>
### <span style="color:pink"> Chaque ligne correspond bien à 1 seul pays car la fréquence max est 1 pour tous les champs</span>
### <span style="color:pink"> 2-alpha code: 238 valeurs sur 241: 3 doublons ou 3 valeurs manquantes</span>
### <span style="color:pink"> Il y a 215 monnaie,152 monnaies uniques référencés manque 26 devises</span>
### <span style="color:pink"> La monnai la plus fréquente est l'euro 26 fois présente</span>
### <span style="color:pink"> On a 214 région sur 241 pays, dont 7 différentes j'en conclue que les pays sont situé sur de grosses zones géographiques</span>
### <span style="color:pink"> La région la plus courante est Europe & Central Asia mentionné 57 fois</span>
### <span style="color:pink">214 revenus au total, classés en 7 catégories pour les 241 pays,les catégories sont Low, Lower middle, Upper middle, High income OECD, High income non-OECD</span>
### <span style="color:pink">Upper middle income soit revenue moyen supérieur est le groupe le plus présent 55X</span>
### <span style="color:pink">Pour l'année des comptes nationaux, on a 205 lignes d'années, 43 années différentes, la plus présente est 2005 34 fois</span>
### <span style="color:pink">Statut de déclaration de la dette extérieur 124 au totale 3 catégorie, le plus présent est Actuel présent 107X 200 systèmes commerciales au global, classé en 2 catégories, la plus présente général trade system présente 106X</span>
### <span style="color:pink">Apres on a des renseignements sur le concept de comptabilité commerciale gouv 2 catégories</span>
### <span style="color:pink">On a des infos sur les norme de diffusion des données du FMI, 2 présents, le générale est plus présent</span>
### <span style="color:pink">On a des infos sur le dernier recensement des donnés, présent pour 213 pays 28 manquants, l'année 2011 est la plus présente 59 pays</span>
### <span style="color:pink">On a des infos sur la dernière enquete des ménages, 60 types d'enquêtes</span>
### <span style="color:pink">On a des infos sur une étude effectué sur les revenus et les dépenses, renseigné pour 160 pays, 75 études, celle auprès des ménages est la plus présente X15</span>
### <span style="color:pink">Autres infos, sur l'enregistrement de l'état civil, 110 l'on fait sur 111 enregistré</span>
### <span style="color:pink">Autres infos, sur l'enregistrement de l'agriculture 142 enregistré</span>
### <span style="color:pink">Autres infos, sur le retrait d'eau 179 renseignés</span>

In [17]:
print(df_country['Income Group'].value_counts())

Income Group
Upper middle income     55
Lower middle income     50
High income: nonOECD    44
Low income              34
High income: OECD       31
Name: count, dtype: int64


### <span style="color:pink"> Juste par curiosité je me suis dit qu'il serai intéressant de connaitre le groupe de revenu auquel appartien chaque pays</span>
### <span style="color:pink"> On avait vu avant que le plus présent était Upper middle income dans 55 pays</span>

In [18]:
print(df_country.describe())

       National accounts reference year  Latest industrial data  \
count                          32.00000              107.000000   
mean                         2001.53125             2008.102804   
std                             5.24856                2.616834   
min                          1987.00000             2000.000000   
25%                          1996.75000             2007.500000   
50%                          2002.00000             2009.000000   
75%                          2005.00000             2010.000000   
max                          2012.00000             2010.000000   

       Latest trade data  
count         185.000000  
mean         2010.994595  
std             2.569675  
min          1995.000000  
25%          2011.000000  
50%          2012.000000  
75%          2012.000000  
max          2012.000000  


### <span style="color:pink"> Je descrit pour les donnnés numériques et les objets</span>
### <span style="color:pink"> Il y a 3 infos: </span>
### <span style="color:pink"> Sur l'année de références des comptes nationaux, 32 pays sont renseignés,l'année la plus ancienne 1987 et plus récente 2012</span>
### <span style="color:pink"> Sur les dernières données de l'industrie: 107 sont renseigné,l'année la plus ancienne 2000, plus récente 2010</span>
### <span style="color:pink"> Sur les dernières donnés lié au commerce: 185 sont renseignés, l'année la plus ancienne 1995 et plus récente 2012</span>
### <span style="color:pink"> J'en conclue que les donnés ne sont pas très a jour </span>

In [19]:
occurrences_df_country = df_country['Country Code'].value_counts()
print(occurrences_df_country)

Country Code
ABW    1
LCA    1
MRT    1
MUS    1
MWI    1
      ..
GRL    1
GTM    1
GUM    1
GUY    1
ZWE    1
Name: count, Length: 241, dtype: int64


In [20]:
occurrences_df_country = df_country['Currency Unit'].value_counts()
print(occurrences_df_country)

Currency Unit
Euro                     23
CFA franc                14
U.S. dollar              14
East Caribbean dollar     6
Danish krone              3
                         ..
Georgian lari             1
Fijian dollar             1
Ethiopian birr            1
Eritrean nakfa            1
New Zambian kwacha        1
Name: count, Length: 152, dtype: int64


In [13]:
occurrences_df_country = df_country['Region'].value_counts()
print(occurrences_df_country)

Region
Europe & Central Asia         57
Sub-Saharan Africa            48
Latin America & Caribbean     41
East Asia & Pacific           36
Middle East & North Africa    21
South Asia                     8
North America                  3
Name: count, dtype: int64


In [21]:
occurrences_df_country = df_country['Income Group'].value_counts()
print(occurrences_df_country)

Income Group
Upper middle income     55
Lower middle income     50
High income: nonOECD    44
Low income              34
High income: OECD       31
Name: count, dtype: int64


### <span style="color:pink"> J'ai compté le nombre d'occurence d'indicateurs intéressant peut être pour la suite </span>
### <span style="color:pink">Country Code, Income Group, Region, Currency Unit, Country Code</span>

In [22]:
doublons = df_country.duplicated().sum()
print("Nombre de doublons :", doublons)

Nombre de doublons : 0


### <span style="color:pink">Il n'y a pas de doublons sur ce DF</span>

In [23]:
total_valeurs_manquantes = df_country.isnull().sum().sum()
print("Total de valeurs manquantes :", total_valeurs_manquantes)

Total de valeurs manquantes : 2017


In [24]:
missing = df_country.isnull().sum()
missing = missing[missing > 0]
print(missing)

2-alpha code                                           3
Currency Unit                                         26
Region                                                27
Income Group                                          27
WB-2 code                                              1
National accounts base year                           36
National accounts reference year                     209
SNA price valuation                                   44
Lending category                                      97
Other groups                                         183
System of National Accounts                           26
Alternative conversion factor                        194
PPP survey year                                       96
Balance of Payments Manual in use                     60
External debt Reporting status                       117
System of trade                                       41
Government Accounting concept                         80
IMF data dissemination standard

### <span style="color:pink">Il y a 2017 valeurs manquantes sur mon DF Country, répartis dans ces colonnes</span>

In [25]:
total_rows = len(df_country)

missing_counts = df_country.isnull().sum()

for col, missing in missing_counts.items():
    if missing > 0:
        print(f"{col} : {missing} valeurs manquantes sur {total_rows} lignes → soit {round((missing/total_rows)*100, 2)} %")

2-alpha code : 3 valeurs manquantes sur 241 lignes → soit 1.24 %
Currency Unit : 26 valeurs manquantes sur 241 lignes → soit 10.79 %
Region : 27 valeurs manquantes sur 241 lignes → soit 11.2 %
Income Group : 27 valeurs manquantes sur 241 lignes → soit 11.2 %
WB-2 code : 1 valeurs manquantes sur 241 lignes → soit 0.41 %
National accounts base year : 36 valeurs manquantes sur 241 lignes → soit 14.94 %
National accounts reference year : 209 valeurs manquantes sur 241 lignes → soit 86.72 %
SNA price valuation : 44 valeurs manquantes sur 241 lignes → soit 18.26 %
Lending category : 97 valeurs manquantes sur 241 lignes → soit 40.25 %
Other groups : 183 valeurs manquantes sur 241 lignes → soit 75.93 %
System of National Accounts : 26 valeurs manquantes sur 241 lignes → soit 10.79 %
Alternative conversion factor : 194 valeurs manquantes sur 241 lignes → soit 80.5 %
PPP survey year : 96 valeurs manquantes sur 241 lignes → soit 39.83 %
Balance of Payments Manual in use : 60 valeurs manquantes su

### <span style="color:pink">Je cherche a savoir le nombre de valeurs manquantes par rapport au totale de ligne pour pouvoir faire une analyse</span>
### <span style="color:pink">Le plus grosses lacunes de valeurs manquante se situent sur les 4 colonnes: </span>
### <span style="color:pink">National accounts reference year : 209 valeurs manquantes sur 241 lignes → soit 86.72 %</span>
### <span style="color:pink">Other groups : 183 valeurs manquantes sur 241 lignes → soit 75.93 %</span>
### <span style="color:pink">Alternative conversion factor : 194 valeurs manquantes sur 241 lignes → soit 80.5 %</span>
### <span style="color:pink">En analyse, je me dit que peux êtres ces donnés la ne seront pas exploitable car trop de vide pas assez de donnés?</span>

### <span style="color:violet"> DATA FRAME 'DATA'</span> 

In [37]:
df_data.columns

Index(['Country Name', 'Country Code', 'Indicator Name', 'Indicator Code',
       '1970', '1971', '1972', '1973', '1974', '1975', '1976', '1977', '1978',
       '1979', '1980', '1981', '1982', '1983', '1984', '1985', '1986', '1987',
       '1988', '1989', '1990', '1991', '1992', '1993', '1994', '1995', '1996',
       '1997', '1998', '1999', '2000', '2001', '2002', '2003', '2004', '2005',
       '2006', '2007', '2008', '2009', '2010', '2011', '2012', '2013', '2014',
       '2015', '2016', '2017', '2020', '2025', '2030', '2035', '2040', '2045',
       '2050', '2055', '2060', '2065', '2070', '2075', '2080', '2085', '2090',
       '2095', '2100', 'Unnamed: 69'],
      dtype='object')

In [38]:
print(df_data.shape)

(886930, 70)


In [39]:
print(df_data.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 886930 entries, 0 to 886929
Data columns (total 70 columns):
 #   Column          Non-Null Count   Dtype  
---  ------          --------------   -----  
 0   Country Name    886930 non-null  object 
 1   Country Code    886930 non-null  object 
 2   Indicator Name  886930 non-null  object 
 3   Indicator Code  886930 non-null  object 
 4   1970            72288 non-null   float64
 5   1971            35537 non-null   float64
 6   1972            35619 non-null   float64
 7   1973            35545 non-null   float64
 8   1974            35730 non-null   float64
 9   1975            87306 non-null   float64
 10  1976            37483 non-null   float64
 11  1977            37574 non-null   float64
 12  1978            37576 non-null   float64
 13  1979            36809 non-null   float64
 14  1980            89122 non-null   float64
 15  1981            38777 non-null   float64
 16  1982            37511 non-null   float64
 17  1983      

### <span style="color:pink"> Le DF Data contient 886 930 lignes qui sont numéroté de 0 à 886 929 et 70 colonnes </span>
### <span style="color:pink"> 4 colonnes contiennent du texte ou chaine de caractères </span>
### <span style="color:pink"> 66 colonnes contiennent des chiffres décimales </span>
### <span style="color:pink"> On remarque que la colonne'unnamed'est vide: je choisi de la supprimer tout de suite</span>

In [41]:
df_data = df_data.drop(columns=['Unnamed: 69'])

In [42]:
df_data.head()

Country Name Country Code  \
0   Arab World          ARB   
1   Arab World          ARB   
2   Arab World          ARB   
3   Arab World          ARB   
4   Arab World          ARB   

                                      Indicator Name  Indicator Code  \
0  Adjusted net enrolment rate, lower secondary, ...      UIS.NERA.2   
1  Adjusted net enrolment rate, lower secondary, ...    UIS.NERA.2.F   
2  Adjusted net enrolment rate, lower secondary, ...  UIS.NERA.2.GPI   
3  Adjusted net enrolment rate, lower secondary, ...    UIS.NERA.2.M   
4  Adjusted net enrolment rate, primary, both sex...     SE.PRM.TENR   

        1970       1971       1972       1973       1974      1975  ...  2055  \
0        NaN        NaN        NaN        NaN        NaN       NaN  ...   NaN   
1        NaN        NaN        NaN        NaN        NaN       NaN  ...   NaN   
2        NaN        NaN        NaN        NaN        NaN       NaN  ...   NaN   
3        NaN        NaN        NaN        NaN        NaN       NaN  ...   NaN   
4  54.822121  54.894138  56.209438  57.267109  57.991138  59.36554  ...   NaN   

   2060  2065  2070  2075  2080  2085  2090  2095  2100  
0   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN  
1   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN  
2   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN  
3   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN  
4   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN  

[5 rows x 69 columns]

In [43]:
print(df_data.describe(include='object'))

       Country Name Country Code  \
count        886930       886930   
unique          242          242   
top      Arab World          ARB   
freq           3665         3665   

                                           Indicator Name Indicator Code  
count                                              886930         886930  
unique                                               3665           3665  
top     Adjusted net enrolment rate, lower secondary, ...     UIS.NERA.2  
freq                                                  242            242  


### <span style="color:pink"> Dans ce DF, il n'y a que 4 colonnes qui contiennent du texte ou chaine de caractères </span>
### <span style="color:pink">La colonne Country Name et Code, contiennt 886 930 lignes, dont 242 uniques: les différents noms de pays</span>
### <span style="color:pink">Le nom de pays le plus fréquent est Arab Word présent 3 665X, jeu de données assez complet</span>
### <span style="color:pink">La colonne indicator Name et Code contiennt elles 2, 886930 lignes dont 3665 indicateurs différents</span>
### <span style="color:pink">L'indicateur le plus présent est Taux net ajusté de scolarisation, premier cycle du secondaire, les deux sexes 242 fois en proportion</span>

In [44]:
print(df_data.describe())

               1970          1971          1972          1973          1974  \
count  7.228800e+04  3.553700e+04  3.561900e+04  3.554500e+04  3.573000e+04   
mean   1.974772e+09  4.253638e+09  4.592365e+09  5.105006e+09  5.401493e+09   
std    1.211687e+11  1.804814e+11  1.914083e+11  2.059170e+11  2.112150e+11   
min   -1.435564e+00 -1.594625e+00 -3.056522e+00 -4.032582e+00 -4.213563e+00   
25%    8.900000e-01  8.853210e+00  9.240920e+00  9.595200e+00  9.861595e+00   
50%    6.317724e+00  6.316240e+01  6.655139e+01  6.969595e+01  7.087760e+01   
75%    6.251250e+01  5.655200e+04  5.863650e+04  6.202900e+04  6.383675e+04   
max    1.903929e+13  1.986457e+13  2.100916e+13  2.238367e+13  2.282991e+13   

               1975          1976          1977          1978          1979  \
count  8.730600e+04  3.748300e+04  3.757400e+04  3.757600e+04  3.680900e+04   
mean   2.314288e+09  5.731808e+09  6.124437e+09  6.671489e+09  7.436724e+09   
std    1.375059e+11  2.215546e+11  2.325489e+11  2.

### <span style="color:pink"> Croissance énorme des valeurs moyennes jusqu’aux années 1980 puis baisse, reprise et stabilisation jusqu'a 2100 </span>
### <span style="color:pink">Présence de valeurs aberrantes, valeurs minimales négatives: -4.21 en 1974, -3.65 en 1975, alors que la plupart des indicateurs comme la scolarisation ou la population ne devraient jamais être négatifs</span>
### <span style="color:pink">Chute de la moyenne après 2000, à partir de 2055, les moyennes deviennent très faibles, alors qu'on avait des milliards auparavant</span>
### <span style="color:pink">2095 sont des prévisions/projections?</span>

In [46]:
doublons = df_data.duplicated().sum()
print("Nombre de doublons :", doublons)

Nombre de doublons : 0


### <span style="color:pink"> Pas de doublons dans ce DF  </span>

In [47]:
total_manquantes = df_data.isnull().sum().sum()
print("Total de valeurs manquantes :", total_manquantes)

Total de valeurs manquantes : 52568249


In [48]:
missing = df_data.isnull().sum()
missing = missing[missing > 0]
print(missing)

1970    814642
1971    851393
1972    851311
1973    851385
1974    851200
         ...  
2080    835494
2085    835494
2090    835494
2095    835494
2100    835494
Length: 65, dtype: int64


### <span style="color:pink"> Il y a 52568249 valeurs manquante dans ce DF, le plus est l'année 1971, trop de valeurs manquantes signifie que les donnés ne sont peut être pas exploitables  </span>

### <span style="color:violet"> DATA FRAME FOOT</span> 

In [3]:
df_foot.columns

Index(['CountryCode', 'SeriesCode', 'Year', 'DESCRIPTION', 'Unnamed: 4'], dtype='object')

In [4]:
print(df_foot.shape)

(643638, 5)


In [5]:
print(df_foot.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 643638 entries, 0 to 643637
Data columns (total 5 columns):
 #   Column       Non-Null Count   Dtype  
---  ------       --------------   -----  
 0   CountryCode  643638 non-null  object 
 1   SeriesCode   643638 non-null  object 
 2   Year         643638 non-null  object 
 3   DESCRIPTION  643638 non-null  object 
 4   Unnamed: 4   0 non-null       float64
dtypes: float64(1), object(4)
memory usage: 24.6+ MB
None


### <span style="color:pink"> Le DF Data contient 643 638 lignes qui sont numéroté de 0 à 643 637 et 5 colonnes </span>
### <span style="color:pink"> 4 colonnes contiennent du texte ou chaine de caractères </span>
### <span style="color:pink"> 1 colonne ds chiffres décimales mais on remarque qu'il s'agit de la colonne'unnamed'qui est vide: je choisi de la supprimer tout de suite</span>

In [8]:
df_foot = df_foot.drop(columns=['Unnamed: 4'])

In [9]:
df_foot.head()

CountryCode      SeriesCode    Year          DESCRIPTION
0         ABW  SE.PRE.ENRL.FE  YR2001  Country estimation.
1         ABW  SE.TER.TCHR.FE  YR2005  Country estimation.
2         ABW  SE.PRE.TCHR.FE  YR2000  Country estimation.
3         ABW  SE.SEC.ENRL.GC  YR2004  Country estimation.
4         ABW     SE.PRE.TCHR  YR2006  Country estimation.

In [10]:
print(df_foot.describe(include='object'))

       CountryCode   SeriesCode    Year   DESCRIPTION
count       643638       643638  643638        643638
unique         239         1558      56          9102
top            LIC  SH.DYN.MORT  YR2004  Country Data
freq          7320         9226   27128        191188


### <span style="color:pink"> Dans ce DF, il n'y a que 4 colonnes qui contiennent du texte ou chaine de caractères </span>
### <span style="color:pink">La colonne Country Code, contient 643 638 lignes, dont 239 uniques: les différents noms de pays</span>
### <span style="color:pink">Le country code le plus fréquent est LIC présent 7 320X peut être Low Income Countries</span>
### <span style="color:pink">La colonne indicator Series Code contient 643 638 lignes, dont 1 558 indicateurs différents, le plus présente est SH.DYN.MORT présent 9 226X, utilisé pour la mortalité infantile </span>
### <span style="color:pink">La colonne Year contient 643 638 lignes, dont 56 indicateurs différents, le plus présente est l'année 2004 présente 27 128 X </span>
### <span style="color:pink">La colonne DESCRIPTION contient 643 638 lignes, dont 9102 indicateurs différents, le plus présente est Country Data présent 191 188 X </span>

In [11]:
occurrences_df_foot = df_foot['SeriesCode'].value_counts()
print(occurrences_df_foot)

SeriesCode
SH.DYN.MORT                 9226
SE.PRM.AGES                 8771
SE.PRM.DURS                 8771
SE.SEC.DURS                 8619
SE.SEC.AGES                 8581
                            ... 
SABER.STD.ASS.GOAL1.LVL2       1
SABER.STD.ASS.GOAL2            1
SABER.STD.ASS.GOAL2.LVL1       1
SABER.STD.ASS.GOAL2.LVL2       1
UIS.XSPENDP.1.FDPUB.FNS        1
Name: count, Length: 1558, dtype: int64


In [13]:
occurrences_df_foot = df_foot['CountryCode'].value_counts()
print(occurrences_df_foot)

CountryCode
LIC    7320
CYP    7183
LDC    6481
SSA    6389
SSF    6336
       ... 
IMN      32
CUW      23
XKX      12
MNP       4
FRO       2
Name: count, Length: 239, dtype: int64


### <span style="color:pink">Je compte les occurences de 2 colonnes par curiosité et comparatif avec les autres tableaux </span>

In [17]:
doublons = df_foot.duplicated().sum()
print("Nombre de doublons :", doublons)

Nombre de doublons : 0


### <span style="color:pink"> Pas de doublons dans ce DF  </span>

In [18]:
total_manquantes = df_foot.isnull().sum().sum()
print("Total de valeurs manquantes :", total_manquantes)

Total de valeurs manquantes : 0


### <span style="color:pink"> Pas de valeurs manquante dans ce DF  </span>

### <span style="color:violet"> DATA FRAME FOOT</span> 

In [37]:
df_stat_series.columns

Index(['Series Code', 'Topic', 'Indicator Name', 'Short definition',
       'Long definition', 'Unit of measure', 'Periodicity', 'Base Period',
       'Other notes', 'Aggregation method', 'Limitations and exceptions',
       'Notes from original source', 'General comments', 'Source',
       'Statistical concept and methodology', 'Development relevance',
       'Related source links', 'Other web links', 'Related indicators',
       'License Type', 'Unnamed: 20'],
      dtype='object')

In [38]:
print(df_stat_series.shape)

(3665, 21)


In [39]:
print(df_stat_series.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3665 entries, 0 to 3664
Data columns (total 21 columns):
 #   Column                               Non-Null Count  Dtype  
---  ------                               --------------  -----  
 0   Series Code                          3665 non-null   object 
 1   Topic                                3665 non-null   object 
 2   Indicator Name                       3665 non-null   object 
 3   Short definition                     2156 non-null   object 
 4   Long definition                      3665 non-null   object 
 5   Unit of measure                      0 non-null      float64
 6   Periodicity                          99 non-null     object 
 7   Base Period                          314 non-null    object 
 8   Other notes                          552 non-null    object 
 9   Aggregation method                   47 non-null     object 
 10  Limitations and exceptions           14 non-null     object 
 11  Notes from original source    

### <span style="color:pink"> Le DF Data contient 3 665 lignes qui sont numéroté de 0 à 3 664 et 21 colonnes </span>
### <span style="color:pink"> 15 colonnes contiennent du texte ou chaine de caractères </span>
### <span style="color:pink"> 6 colonnes des chiffres décimaux mais on remarque que 6 d'entre eux sont vides: je choisi de la supprimer tout de suite</span>

In [40]:
df_stat_series = df_stat_series.drop(columns=['Unnamed: 20'])

In [41]:
df_stat_series = df_stat_series.drop(columns=['License Type'])

In [42]:
df_stat_series = df_stat_series.drop(columns=['Related indicators'])

In [43]:
df_stat_series = df_stat_series.drop(columns=['Other web links'])

In [44]:
df_stat_series = df_stat_series.drop(columns=['Notes from original source'])

In [45]:
df_stat_series = df_stat_series.drop(columns=['Unit of measure'])

In [46]:
df_stat_series.head()

Series Code       Topic  \
0  BAR.NOED.1519.FE.ZS  Attainment   
1     BAR.NOED.1519.ZS  Attainment   
2  BAR.NOED.15UP.FE.ZS  Attainment   
3     BAR.NOED.15UP.ZS  Attainment   
4  BAR.NOED.2024.FE.ZS  Attainment   

                                      Indicator Name  \
0  Barro-Lee: Percentage of female population age...   
1  Barro-Lee: Percentage of population age 15-19 ...   
2  Barro-Lee: Percentage of female population age...   
3  Barro-Lee: Percentage of population age 15+ wi...   
4  Barro-Lee: Percentage of female population age...   

                                    Short definition  \
0  Percentage of female population age 15-19 with...   
1  Percentage of population age 15-19 with no edu...   
2  Percentage of female population age 15+ with n...   
3  Percentage of population age 15+ with no educa...   
4  Percentage of female population age 20-24 with...   

                                     Long definition Periodicity Base Period  \
0  Percentage of female population age 15-19 with...         NaN         NaN   
1  Percentage of population age 15-19 with no edu...         NaN         NaN   
2  Percentage of female population age 15+ with n...         NaN         NaN   
3  Percentage of population age 15+ with no educa...         NaN         NaN   
4  Percentage of female population age 20-24 with...         NaN         NaN   

  Other notes Aggregation method Limitations and exceptions General comments  \
0         NaN                NaN                        NaN              NaN   
1         NaN                NaN                        NaN              NaN   
2         NaN                NaN                        NaN              NaN   
3         NaN                NaN                        NaN              NaN   
4         NaN                NaN                        NaN              NaN   

                                              Source  \
0  Robert J. Barro and Jong-Wha Lee: http://www.b...   
1  Robert J. Barro and Jong-Wha Lee: http://www.b...   
2  Robert J. Barro and Jong-Wha Lee: http://www.b...   
3  Robert J. Barro and Jong-Wha Lee: http://www.b...   
4  Robert J. Barro and Jong-Wha Lee: http://www.b...   

  Statistical concept and methodology Development relevance  \
0                                 NaN                   NaN   
1                                 NaN                   NaN   
2                                 NaN                   NaN   
3                                 NaN                   NaN   
4                                 NaN                   NaN   

  Related source links  
0                  NaN  
1                  NaN  
2                  NaN  
3                  NaN  
4                  NaN

In [47]:
df_stat_series = df_stat_series.drop(columns=['Source'])

In [50]:
df_stat_series = df_stat_series.drop(columns=['Related source links'])

### <span style="color:pink"> Je supprime la colonne source et link qui je suis sur ne me servira pas </span>

In [51]:
print(df_stat_series.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3665 entries, 0 to 3664
Data columns (total 13 columns):
 #   Column                               Non-Null Count  Dtype 
---  ------                               --------------  ----- 
 0   Series Code                          3665 non-null   object
 1   Topic                                3665 non-null   object
 2   Indicator Name                       3665 non-null   object
 3   Short definition                     2156 non-null   object
 4   Long definition                      3665 non-null   object
 5   Periodicity                          99 non-null     object
 6   Base Period                          314 non-null    object
 7   Other notes                          552 non-null    object
 8   Aggregation method                   47 non-null     object
 9   Limitations and exceptions           14 non-null     object
 10  General comments                     14 non-null     object
 11  Statistical concept and methodology  23 non

### <span style="color:pink"> Juste pour controler que j'ai que des objects </span>

In [53]:
print(df_stat_series.describe(include='object'))

                Series Code              Topic  \
count                  3665               3665   
unique                 3665                 37   
top     BAR.NOED.1519.FE.ZS  Learning Outcomes   
freq                      1               1046   

                                           Indicator Name  \
count                                                3665   
unique                                               3665   
top     Barro-Lee: Percentage of female population age...   
freq                                                    1   

                                         Short definition  \
count                                                2156   
unique                                               1169   
top     Data Interpretation: 1=Latent; 2=Emerging; 3=E...   
freq                                                  215   

                                          Long definition Periodicity  \
count                                                3665       

### <span style="color:pink">La colonne Series Code contient 3665 indicateurs </span>
### <span style="color:pink">Tous aparaissent 1X par exemple l'indicateur BAR.NOED.1519.FE.ZS, j'imagine sur la scolarisation des femmes </span>
### <span style="color:pink">Il y a 37 thèmes, Learning Outcomes est le plus fréquent avec 1046 X</span>
### <span style="color:pink">Chaque code possède un nom nunique 3665</span>
### <span style="color:pink">2156 definition courtes 1169 catégories</span>
### <span style="color:pink">3665 Long definition Periodicity dont 2060 catégories, Data Interpretation: 1=Latent; 2=Emerging; 3=E aprait le plus 215X</span>
### <span style="color:pink">Une colonne indique si les données concernent des projections ou des périodes spécifiques: Projections (2010 to 2100)</span>
### <span style="color:pink"> les dernières colonnes Aggregation method est très peu renseigné 47 lignes dont 3 uniques</span>
### <span style="color:pink">Limitations and exceptions très peu renseigné 14 dont 8 unique</span>